cli/pcluster/createami.py

```diff
@@ -111,6 +113,49 @@ def _get_cookbook_dir(region, template_url, args, tmpdir):
         sys.exit(1)
 
 
+def _is_valid_post_install_script(post_install_script_url):
+    return any(
+        [
+            post_install_script_url.startswith("https://"),
+            post_install_script_url.startswith("s3://"),
+            post_install_script_url.startswith("file://"),
+        ]
+    )
+
+
+def _get_current_timestamp():
+    return datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+
+
+def _get_post_install_script_dir(post_install_script_url, tmp_dir):
+    LOGGER.info("Post install script url: %s", post_install_script_url)
+    try:
+        tmp_post_install_script_folder = os.path.join(tmp_dir, "script")
+        os.mkdir(tmp_post_install_script_folder)
+        if post_install_script_url is not None:
+            if not _is_valid_post_install_script(post_install_script_url):
+                raise URLError("Invalid URL")
+            tmp_post_install_script_path = os.path.join(
+                tmp_post_install_script_folder, _get_current_timestamp() + "-" + post_install_script_url.split("/")[-1]
+            )
+            if post_install_script_url.startswith("https"):
+                urlretrieve(post_install_script_url, filename=tmp_post_install_script_path)
+            elif post_install_script_url.startswith("s3"):
+                s3_client = boto3.client("s3")
+                output = urlparse(post_install_script_url)
+                s3_client.download_file(output.netloc, output.path.lstrip("/"), tmp_post_install_script_path)
+            elif post_install_script_url.startswith("file"):
+                copyfile(post_install_script_url.replace("file://", ""), tmp_post_install_script_path)
+        else:
+            tmp_post_install_script_path = None
+        LOGGER.info("Post install script dir: %s", tmp_post_install_script_path)
```

Comment:
Here we want to pass a `None` parameter as post_install_url to the `run packer` command because of consistency. We allowed the case like users create ami without post_install_script. And in `packer_{os}.json`. we also upload script folder to `/tmp/createami_script`, but do nothing if createami_script is empty, then provision will clean the `/tmp` directory.  It is the easiest way I find to keep with/without `post_install_script` in the same workflow.
But in `Logger.info`, maybe there is a better way to print the empty post_install_script rather than `None`

Reply:
You can write something like:
```python
LOGGER.info("Post install script dir %s", tmp_post_install_script_path if tmp_post_install_script_path else "not specified.")
```

Reply:
Done